R/load_precomputed_layouts.R

```diff
@@ -45,6 +45,13 @@ ReadMPX_layouts <- function (
 
   # Create temporary directory
   temp_layout_dir <- file.path(fs::path_temp(), "temp_layouts")
+  while (fs::dir_exists(temp_layout_dir)) {
+    err <- try(fs::dir_delete(temp_layout_dir))
+    if (inherits(err, "try-error")) {
+      warn(glue("Failed to delete temporary directory '{col_br_blue(temp_layout_dir)}'."))
+      temp_layout_dir <- file.path(fs::path_temp(), "temp_layouts", .generate_random_string())
+    }
+  }
```

Comment:
I don't fully understand this part. Do you want to remove the temporary folder if it exists in order to create a path that is not used? If so, think this could possibly be a separate utility function for readability. I would suspect we would want to use it in other places.

Reply:
So what happens is that we generate a new temp folder with `file.path(fs::path_temp(), "temp_layouts", .generate_random_string())`. 

`.generate_random_string()` creates a random string with 5 letters/integers. This is just to ensure that we use an unique name to create a new empty directory. However, if we reset the seed there's a risk that we create a folder with the same name. If this folder contains layout parquet files these will also be loaded which is something we want to avoid. Usually, this shouldn't be an issue, but can happen for example when we run the tests where we reset the seed quite frequently.

Another annoying thing is that there's no guarantee that we can actually delete files in the temp folder. When we run the R CMD check on windows (github workflow) this is the case. So what I have attempted here is to create a new temporary directory name, delete it if it already exists, but if we cannot delete it we try to create a new name that is not being used. This is mainly to make sure that everything works on our github workflow (windows) but in general, the folder should be deleted if it already exists and created again. Once we have loaded the layouts, we don't need the folder anymore so it should be safe to remove.

If there's a cleaner way to do this, that would be great!